### DISCUSSION TITLE
"Expand card" button to expand a Card

### DISCUSSION THREAD

**Discussion by RenataZakauske** [Shopify/polaris-react#7006] [Ideas] [answered]
## Feature request summary
"Expand card" button could be added to the bottom of a card. When added, the "Expand card" button is fixed to the bottom of a card. When clicked, it expands the card to show more relevant content within the card. Having a shared component would allow for convince and consistency.

Suggestion:
![image](https://user-images.githubusercontent.com/107629461/184929123-362d1078-a6c3-4ef7-b312-e9510c79eaa9.png)

Hover state: 
![image](https://user-images.githubusercontent.com/107629461/184929275-b7dc5485-b3c0-4830-a8ff-a9f8fd23c5d9.png)

## Usage
Currently being used on Product pages, Customer pages and Forms (1P App).

## Rationale
Product and B2B Customer pages both use it but are not implemented the same way and do not have same UI interaction patterns. Specifically, hover states are not the same. Currently there isn’t a consistent button for expanding cards. The alternatives is using “Collapsible” which is a component that can be shown or hidden. The“Collapsible” component does not provide a recommendation on a visual UI to use.

![image](https://user-images.githubusercontent.com/107629461/185098256-3f2120b2-495f-4092-8f57-6dc1139fa12c.png)

![image](https://user-images.githubusercontent.com/107629461/185101074-b28c46be-4783-41c3-8996-ec2bc149cc97.png)

Link to the video, where this button is being used:
https://shopify.enterprise.slack.com/files/U03NV9XTRTK/F03U21YBZFW/screen_recording_2022-08-09_at_8.21.46_am.mov

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s [:+1: upvotes](https://help.github.com/articles/about-conversations-on-github/) to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md) for more information.

**Comment by ghost** [comment]
👋 Thanks for opening your first issue. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris/blob/master/.github/CONTRIBUTING.md).

**Comment by yurm04** [comment, marked as answer]
Hey @RenataZakauske 👋 thanks for reaching out. We don't currently have plans to add this feature to the Card component. However, we welcome you to contribute this change to the system if there is a merchant need for it in multiple places across the admin. I recommend reaching out in the internal #polaris and #admin-ux slack channels, organizing a [Polaris Lab](https://ux.shopify.com/how-to-push-the-limits-of-a-design-system-b44fac420be) with the relevant teams, and working with our team to support on any systems related topics.

Please use this discussion to facilitate conversation and get feedback from other UXers

**Reply by idonahue** [reply]
Hi @yurm04 , thanks for the reply! 

I am the UXer on @RenataZakauske 's team. From a UX perspective, I think it's a solid choice and makes sense to add to Polaris. We got the idea from Admin (currently 2 instances are being used, which are listed above.  We will be the 3rd place in admin using it. 

They they both visually look the same but since they are both custom components, they do not function the same (unfortunately). @RenataZakauske built a component and we would like to contribute it to Polaris. Neither of us have contributed (Code or Figma component) to Polaris before. 

How should we proceed? Should @RenataZakauske do a pull request and I can then branch Polaris' Figma and add the component?

**Reply by yurm04** [reply]
@idonahue I checked with the developers on the team and it turns out there is a way to recreate this functionality using existing Polaris components. Check out [this codesandbox demo](https://codesandbox.io/s/brave-panini-smjpf4?file=/App.js) and the attached video.

https://user-images.githubusercontent.com/4642404/187531565-9b993add-53e8-4a82-96e2-c1cb7b7daed0.mov

You can achieve this by using the `Card`, `Collapsible`, and `Button` components. The [`Button`](https://polaris.shopify.com/components/button) component has a `disclosure` prop that can be used.

Instead of making changes to the card component, you can create a global component in the [shopify/web](https://github.com/Shopify/web/tree/main/app/components) using these Polaris elements that manages the show/hide functionality. Then, that new global component can be used anywhere in the shopify/web repo
